### PR TITLE
Add package.json and dev tooling

### PR DESCRIPTION
**Pull request by adroitwhiz**
This adds a `package.json` and a bunch of dev tooling for ESLint and Prettier integration, as well as setting up ESLint with its recommended settings. It also adds editor configuration files for proper formatting.

This shouldn't be merged until a proper license is figured out to put in `package.json`.

### CODE REVIEW

File: scratch-js/Project.mjs [hunk 1]
```diff
@@ -162,6 +162,7 @@ export default class Project {
       };
       audio.addEventListener("ended", soundEnd);
       audio.addEventListener("pause", soundEnd);
+      audio.addEventListener("error", reject);
```
Comment (adroitwhiz) [inline scratch-js/Project.mjs]: This fixes a linter error stating that the promise executor's `reject` argument was never used.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/PullJosh/scratch-js.git"
+  },
+  "author": "Josh Pullen",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/PullJosh/scratch-js/issues"
+  },
+  "homepage": "https://github.com/PullJosh/scratch-js#readme",
```
Comment (PullJosh) [inline package.json]: Should we go with https://scratch-js.now.sh/?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "lint": "eslint 'scratch-js/**'"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/PullJosh/scratch-js.git"
+  },
+  "author": "Josh Pullen",
+  "license": "ISC",
```
Comment (PullJosh) [inline package.json]: Let's put `"LGPL-3.0-or-later"` here unless someone has a better recommendation.

**Comment by PullJosh** [comment]
Eslint/prettier config look beautiful! I always have a bit of trouble getting that right, so good work. :P

File: package.json (hunk 3 above)
Comment (PullJosh) [reply package.json]: Actually, maybe `"MIT"` is the move.

Comment (adroitwhiz) [reply package.json]: I believe the LGPL permits usage of linked libraries, but if you e.g. include `scratch-js` in, say, your game project with a bundler, that's a gray area.

Comment (PullJosh) [reply package.json]: Does spending too long thinking about licenses count as [bikeshedding](https://en.wiktionary.org/wiki/bikeshedding)? :P

**Comment by PullJosh** [comment]
Thanks! :)

**Comment by PullJosh** [comment]
Fixes #31 (will this still work after merging?)

**Pull request merged**